[Regression] [Linux] Build fails due to FAudio/SDL

### Quick summary

Build fails due to an issue with FAudio, which is apparently triggered from an outdated SDL. 

### Details

Log excerpt below.  System libsdl2-dev is indeed older (see below config), however shouldn't the updated 3rdparty SDL2 version be used in this case?
```
[795/2399] Building C object 3rdparty/FAudio/CMakeFiles/FAudio.dir/src/FAudio_platform_sdl2.c.o
FAILED: 3rdparty/FAudio/CMakeFiles/FAudio.dir/src/FAudio_platform_sdl2.c.o 
ccache /usr/bin/cc -DFAUDIO_DISABLE_DEBUGCONFIGURATION -I/home/testing/Desktop/rpcs3/rpcs3/3rdparty/FAudio/src -I/home/testing/Desktop/rpcs3/rpcs3/3rdparty/FAudio/include -I/usr/include/SDL2 -w -O3 -DNDEBUG -std=c99 -MD -MT 3rdparty/FAudio/CMakeFiles/FAudio.dir/src/FAudio_platform_sdl2.c.o -MF 3rdparty/FAudio/CMakeFiles/FAudio.dir/src/FAudio_platform_sdl2.c.o.d -o 3rdparty/FAudio/CMakeFiles/FAudio.dir/src/FAudio_platform_sdl2.c.o -c /home/testing/Desktop/rpcs3/rpcs3/3rdparty/FAudio/src/FAudio_platform_sdl2.c
/home/testing/Desktop/rpcs3/rpcs3/3rdparty/FAudio/src/FAudio_platform_sdl2.c:34:2: error: #error "SDL version older than 2.24.0"
   34 | #error "SDL version older than 2.24.0"
      |  ^~~~~
[812/2399] Building CXX object 3rdparty/llvm_build/lib/MC/MCParser/CMakeFiles/LLVMMCParser.dir/AsmParser.cpp.o
ninja: build stopped: subcommand failed.
```

### Build with regression

Most likely stems from commit 04d9693b51fe8779d5f64bf14a6ccd678e70ad21

### Attach two log files

Logfile: [build.log](https://github.com/RPCS3/rpcs3/files/10609434/build.log)


### Attach capture files for visual issues

N/A

### System configuration

* OS: Linux Mint 21.1 / kernel 5.15
* CPU: Intel I7-10870H
* GPU: Nvidia GTX 3070 / driver 525.78
* libsdl version: `libsdl2-dev/jammy-updates,now 2.0.20+dfsg-2ubuntu1.22.04.1 amd64`
* Commit tested: c0243a92d05191285f20ef22c71fd209c9116a24
* Last known good commit: c5b25fa132d8e5d0b246dea43a769e7f176885a0

### Other details

_No response_

--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -222,21 +222,28 @@ set(FAUDIO_TARGET 3rdparty_dummy_lib)
 if(USE_FAUDIO)
 	# FAudio depends on SDL2
 	find_package(SDL2)
-	if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.0.12)
-		message(WARNING
-			"-- RPCS3: FAudio requires SDL 2.0.9 or newer. Please note, this warning"
-			"can also be displayed with SDL2 versions between 2.0.9-2.0.12, as the"
-			"CMake config files are not correctly installed. Since a valid SDL2"
-			">=2.0.9 version cannot be found, building with FAudio will be skipped.")
-		set(USE_FAUDIO False)
-	else()
-		if (USE_SYSTEM_FAUDIO)
+	if (USE_SYSTEM_FAUDIO)
+		if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.0.12)
+			message(WARNING
+				"-- RPCS3: System FAudio requires SDL 2.0.9 or newer. Please note, this warning"
+				"can also be displayed with SDL2 versions between 2.0.9-2.0.12, as the"
+				"CMake config files are not correctly installed. Since a valid SDL2"
+				">=2.0.9 version cannot be found, building with FAudio will be skipped.")
+			set(USE_FAUDIO False)
+		else()
 			message(STATUS "RPCS3: Using system FAudio")
 			find_package(FAudio REQUIRED CONFIGS FAudioConfig.cmake FAudio-config.cmake)
 			add_library(3rdparty_FAudio INTERFACE)
 			target_link_libraries(3rdparty_FAudio INTERFACE FAudio)
 			target_compile_definitions(3rdparty_FAudio INTERFACE -DHAVE_FAUDIO)
 			set(FAUDIO_TARGET 3rdparty_FAudio)
+		endif()
+	else()
+		if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.24.0)
+			message(WARNING
+				"-- RPCS3: 3rdparty FAudio requires SDL 2.24.0 or newer. Since a valid SDL2"
+				">=2.24.0 version cannot be found, building with FAudio will be skipped.")
+			set(USE_FAUDIO False)
 		else()
 			message(STATUS "RPCS3: Using builtin FAudio")
 			set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library")

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -23,7 +23,7 @@ These are the essentials tools to build RPCS3 on Linux. Some of them can be inst
 * [CMake 3.16.9+](https://www.cmake.org/download/)
 * [Qt 5.15.2](https://www.qt.io/download-qt-installer)
 * [Vulkan SDK 1.3.224+](https://vulkan.lunarg.com/sdk/home) (See "Install the SDK" [here](https://vulkan.lunarg.com/doc/sdk/latest/linux/getting_started.html))
-* [SDL2](https://www.libsdl.org/download-2.0.php) (for the FAudio backend)
+* [SDL2](https://github.com/libsdl-org/SDL/releases) (for the FAudio backend)
 
 **If you have an NVIDIA GPU, you may need to install the libglvnd package.**
 

